frontend/rust-lib/flowy-search/src/error.rs

```diff
@@ -0,0 +1,12 @@
+use flowy_error::{ErrorCode, FlowyError};
+use tantivy::TantivyError;
+
+trait ConvertToFlowyError {
+  fn convert_to_flowy_error(self) -> FlowyError;
```

Comment:
How about adding a feature flag for converting. Just like these:
<img width="267" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/86001920/f32f9021-952b-4a99-b594-bed1e1b15a0f">

Reply:
You can use thiserror to customize your error type. it's very handy

Reply:
Already did add a flag for tantivy support, just forgot to delete this it seems.